### PR TITLE
Батанина - Лабораторная работа#3

### PR DESCRIPTION
**Pull request by SolarSystem23**

### CODE REVIEW

File: modules/matrix-calculator/include/matrix_operation.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#include <string>
+#include <vector>
+
+#include "include/matrix_calculator.h"
+
+class MatrixOperation {
+ public:
+  MatrixOperation();
+  std::string operator()(int argc, const char** argv);
+  std::vector<matrix_type> getInputMatrix();
```
Comment (TolyaTalamanov) [inline modules/matrix-calculator/include/matrix_operation.hpp]: Метод возвращает вектор матриц, а не одну, переименуйте в getInputsMatrix

Comment (SolarSystem23) [reply modules/matrix-calculator/include/matrix_operation.hpp]: наверно вы имели в виду  getInputMatrixes

File: modules/matrix-calculator/include/matrix_calculator.h [hunk 2]
```diff
@@ -24,9 +25,15 @@ class MatrixCalculator {
...
-    matrix_type matrix;
     MatrixCalculator Prepare_for_minores(int, matrix_type, int, int) const;
     MatrixCalculator Transpon(int a) const;
+
+    friend std::ostream& operator<<(std::ostream& stream,
+                             const MatrixCalculator& result);
+    std::string matrix_to_string();
+
+ private:
+    matrix_type matrix;
```
Comment (TolyaTalamanov) [inline modules/matrix-calculator/include/matrix_calculator.h]: m_matrix ?

File: modules/matrix-calculator/src/matrix_operation.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        //   "  $ " + appname + " <rows> " + "<columns> " +
+        //   "<operation>\n\n" +
+        //
+        //   "Where 1 and 2 arguments are int numbers, " +
+        //   "and <operation> is one of" +
+        //       " '+', '-', '*', 'Transpon', 'InverseMatrix'.\n";
+}
+
+int MatrixOperation::find_operation(std::string operation) {
+  std::vector<std::string> operat = {"+", "-", "*",
```
Comment (TolyaTalamanov) [inline modules/matrix-calculator/src/matrix_operation.cpp]: ops ?

**Comment by SolarSystem23** [comment]
@FedorovaOlga

**Comment by FedorovaOlga** [comment]
With error "ERROR: Should be 4 arguments." mention whitch ones. Also user should understand his matrix will be filled randomly. Add this description to this comment.

File: modules/matrix-calculator/src/matrix_operation.cpp [hunk 4]
```diff
@@ -15,23 +15,13 @@ MatrixOperation::MatrixOperation() : message_("") {}
...
-        //+
-        //   "This is a matrix calculator application.\n\n" +
-        //   "Please provide arguments in the following format:\n\n"+
-        //
-        //   "  $ " + appname + " <rows> " + "<columns> " +
-        //   "<operation>\n\n" +
-        //
-        //   "Where 1 and 2 arguments are int numbers, " +
-        //   "and <operation> is one of" +
-        //       " '+', '-', '*', 'Transpon', 'InverseMatrix'.\n";
```
Comment (FedorovaOlga) [inline modules/matrix-calculator/src/matrix_operation.cpp]: Basicly i don't undertand why u have deleted this part. I think it's an important part of application.

Comment (FedorovaOlga) [reply modules/matrix-calculator/src/matrix_operation.cpp]: Which maybe should be handled by case when app is called without any arguments.  But as I've said before list of arguments should be clarified in another errors too.

**Comment by SolarSystem23** [comment]
@TolyaTalamanov @FedorovaOlga  
I fixed all errors.

**Pull request merged**